src/Main/Geo.js

```diff
@@ -1,21 +1,46 @@
-export class GeoBtn extends Component {
-  state = {
-    latitude: null,
-    longitude: null
+import PropTypes from 'prop-types';
+
+export class Geo extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      lat: '',
+      long: ''
+    };
   }
-  getGeoPosition =() => {
+  componentDidMount() {
+    this.setCurrentGeoPosition();
+  }
+
+  setCurrentGeoPosition() {
     navigator.geolocation.getCurrentPosition((position) => {
       const crd = position.coords;
-      this.setState({ latitude: crd.latitude, longitude: crd.longitude });
+      this.setState({ lat: crd.latitude, long: crd.longitude });
+    });
+  }
+
+  getGeoPosition = ({ target }) => {
+    this.setState({
+      [target.name]: target.value
     });
   }
   render() {
+    const { long, lat } = this.state;
+    const src = `http://maps.googleapis.com/maps/api/staticmap?center=${lat},${long}&zoom=12&size=300x300&sensor=false&language=ru&markers=color:blue|label:S|${lat},${long}`;
     return (
-      <div className="geopos">
-        <button onClick={this.getGeoPosition}>Get Position</button>
-        <div>Latitude is: {this.state.latitude}</div>
-        <div>Longitude is {this.state.longitude}</div>
+      <div className="map">
+        <div>
+          <input value={long} onChange={this.getGeoPosition} name="long" type="number" min="-90" max="90" />
+          <input value={lat} onChange={this.getGeoPosition} name="lat" type="number" min="-180" max="180" />
+        </div>
+        <img src={src} alt="" />
       </div>
     );
   }
 }
+
+
+Geo.propTypes = {
+  long: PropTypes.string,
+  lat: PropTypes.string
```

Comment:
again - why is the state in the propTypes? You don't have `long` and `lat` in properties.